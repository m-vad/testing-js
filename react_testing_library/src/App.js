import React, {useEffect, useState} from "react";

const App = () => {
    const [data, setData] = useState(null)

    useEffect( () => {
        setTimeout( () => {
            setData({})
        }, 100)
    }, [])

    return (
        <div>
            {data && <div style={{color: 'red'}}>data</div>}
            <h1>Hello world</h1>
            <button>Click me</button>
            <input type="text" placeholder="input value..."/>
        </div>
    );
}

export default App;
