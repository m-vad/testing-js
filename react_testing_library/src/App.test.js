import { render, screen } from '@testing-library/react';
import App from './App';
import React from "react";

describe('TEST APP', () => {
  test('renders learn react link', () => {
    render(<App />);
    const helloWorldElem = screen.getByText(/hello world/i);
    const btn = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/input value/i);
    expect(helloWorldElem).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    // screen.debug();
    expect(helloWorldElem).toMatchSnapshot();
    expect(btn).toMatchSnapshot();
    expect(input).toMatchSnapshot();
  });

  test('renders learn react link', async () => {
    render(<App />);
    // const helloWorldElem = screen.queryByText(/hello world2/i);
    // expect(helloWorldElem).toBeNull();
    screen.debug();
    const helloWorldElem = await screen.findByText(/data/i);
    expect(helloWorldElem).toBeInTheDocument();
    expect(helloWorldElem).toHaveStyle({color: 'red'});
    screen.debug();
  });
})


