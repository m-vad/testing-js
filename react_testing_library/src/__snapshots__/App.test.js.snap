// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST APP renders learn react link 1`] = `
<h1>
  Hello world
</h1>
`;

exports[`TEST APP renders learn react link 2`] = `
<button>
  Click me
</button>
`;

exports[`TEST APP renders learn react link 3`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<h1>
  Hello world
</h1>
`;

exports[`renders learn react link 2`] = `
<button>
  Click me
</button>
`;

exports[`renders learn react link 3`] = `
<input
  placeholder="input value..."
  type="text"
/>
`;
